src/main/java/com/example/usermanagement/config/SecurityConfig.java

```diff
@@ -1,39 +1,42 @@
 package com.example.usermanagement.config;
 
-import lombok.RequiredArgsConstructor;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.security.authentication.AuthenticationProvider;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.oauth2.server.resource.OAuth2ResourceServerConfigurer;
 import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.session.SessionRegistryImpl;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
-
+import org.springframework.security.web.authentication.session.RegisterSessionAuthenticationStrategy;
+import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy;
 
 @Configuration
 @EnableWebSecurity
-@RequiredArgsConstructor
 public class SecurityConfig {
 
-
-    private final JwtAuthenticationFilter jwtAuthFilter;
-    private final AuthenticationProvider authenticationProvider;
-
+    @Bean
+    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
+        return new RegisterSessionAuthenticationStrategy(new SessionRegistryImpl());
+    }
 
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
-        http.csrf().disable().authorizeHttpRequests(
+        http.
+                csrf().disable()
+                .authorizeHttpRequests(
                         authorize -> authorize
-                                .requestMatchers("/api/v1/authenticate", "/api/v1/register")
+                                .requestMatchers("/api/v1/login")
                                 .permitAll()
                                 .anyRequest()
-                                .authenticated())
-                                .sessionManagement()
-                                .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-                                .and()
-                                .authenticationProvider(authenticationProvider)
-                                .addFilterBefore(jwtAuthFilter, UsernamePasswordAuthenticationFilter.class);
+                                .authenticated());
+        http
+                .oauth2Login();
+        http
+                .oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
+        http
+                .sessionManagement()
+                .sessionCreationPolicy(SessionCreationPolicy.STATELESS);
```

Comment:
check if it will work without sessionManagement or not. why we need it?

Reply:
It will work but it indicates that sessions should not be created or used for authentication and authorization purposes. This approach is commonly used in stateless authentication scenarios, such as token-based authentication or API-only applications.

By setting the session creation policy to Stateless, Spring Security ensures that each request is treated independently without relying on server-side sessions to maintain authentication state. Instead, the authentication is typically performed using tokens, such as JSON Web Tokens (JWT), which are sent in the request headers.